### PR TITLE
Preview on iPad with text description

### LINKED ISSUE #279: Add presentation mode to iPad?

### PR DESCRIPTION
**Pull request by akaDuality**
Closes #279 

- [x] Improve: add StateController that depends on size classes
   - [x] Draw canvas when there is a known trait. The new API is backported to iOS 13
   - [ ] Check state behavior at iOS 16
- [x] Remove jumping animation when Preview's elements have different width
- [x] Add a presentation button to separate drawing and presentation modes like on macOS
- [x] Navigation bar in translucent on iPad 
- [x] Publish changes when text loses focus (and add UI-tests)
- [x] Add image-cache to speed up image loading and test by performance. An image loads whenever the artboard publisher has been changed
- [x] Update elements trait after selection (check by elements color)
   - [ ] Add UI-test for that: expose rectangles for accessiblitiy to use it in UI-tests)

### CODE REVIEW

**Comment by AgapovOne** [comment]
I pushed a fix for building, changed decision from phone to horizontalSizeClass.

But it still doesn't work for OTT samples

<details><summary>Details</summary>
<p>



</p>
</details>

File: Shared/Sources/Presentation/PresentationView.swift [hunk 1]
```diff
@@ -149,11 +149,19 @@ public struct PresentationView: View {
         RoundedRectangle(cornerRadius: 6, style: .continuous)
             .foregroundStyle(
                 { () -> SwiftUI.Color in
+#if os(iOS)
+                    if isControlHovered(control) {
+                        return Color(uiColor: control.color.withAlphaComponent(0.5)) // TODO: Is it fine for iOS?
```
Comment (akaDuality) [inline Shared/Sources/Presentation/PresentationView.swift:154]: Don't know what it better for iOS

File: Shared/Sources/ElementSettings/Components.swift [hunk 2]
```diff
@@ -29,9 +29,10 @@ struct TextValue: View {
     let title: LocalizedStringKey
     @Binding var value: String
     
+    @State var innerValue: String = ""
```
Comment (drucelweisse) [inline Shared/Sources/ElementSettings/Components.swift:32]: is it needed? seems to be unused

File: Shared/Sources/CommonUI/TextFieldOnSubmit.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import SwiftUI
+
+public struct TextFieldOnSubmit: View {
+    @State var label: String
+    @Binding var value: String // Available outside
+    
+    @State var innerValue: String // Used only for drawing inside textField
+    @FocusState private var isFocused: Bool
```
Comment (drucelweisse) [inline Shared/Sources/CommonUI/TextFieldOnSubmit.swift:8]: is private property needed? seems that focus is provided from parent.

If u really want to pass focus here you probably should pass FocusState<Bool>.Binding which is not the same thing as Binding<T>

File: VoiceOver Preview/VoiceOver Preview/BrowserViewController.swift [hunk 4]
```diff
@@ -23,14 +23,22 @@ class DocumentBrowserViewController: UIDocumentBrowserViewController, UIDocument
     }
     
     // MARK: UIDocumentBrowserViewControllerDelegate
-
+#if os(visionOS)
+    func documentBrowser(_ controller: UIDocumentBrowserViewController, didPickDocumentsAt documentURLs: [URL]) {
```
Comment (drucelweisse) [inline VoiceOver Preview/VoiceOver Preview/BrowserViewController.swift:27]: why it's macrossed? do same thing as other function

**Review by drucelweisse — CHANGES_REQUESTED** [review]

File: VoiceOver Preview/VoiceOver Preview/BrowserViewController.swift (hunk 4 above)
Comment (akaDuality) [reply VoiceOver Preview/VoiceOver Preview/BrowserViewController.swift:27]: Nope, `didPickDocumentsAt` vs `didPickDocumentURLs`

Comment (akaDuality) [reply VoiceOver Preview/VoiceOver Preview/BrowserViewController.swift:27]: Who the fuck at Apple did this?

File: Shared/Sources/CommonUI/TextFieldOnSubmit.swift (hunk 3 above)
Comment (akaDuality) [reply Shared/Sources/CommonUI/TextFieldOnSubmit.swift:8]: It's the only way what I found to track lose of focus to publish change of `innerValue`

**Pull request merged**